### ISSUE TITLE
500 Error

### ISSUE THREAD

**Issue by mfaraji** [googleads/googleads-python-lib#176]
I have tried the sample codes "get_first_party_audience_segments" and "get_all_users" with oauth2 token, and I receive 500 errors all the time:

`Traceback (most recent call last):
  File "audience.py", line 78, in <module>
    get_all_users(dfp_client)
  File "audience.py", line 60, in get_all_users
    response = user_service.getUsersByStatement(statement.ToStatement())
  File "/home/ubuntu/venv/local/lib/python2.7/site-packages/googleads/common.py", line 700, in MakeSoapRequest
    *[_PackForSuds(arg, self.suds_client.factory) for arg in args])
  File "/home/ubuntu/venv/local/lib/python2.7/site-packages/suds/client.py", line 521, in __call__
    return client.invoke(args, kwargs)
  File "/home/ubuntu/venv/local/lib/python2.7/site-packages/suds/client.py", line 581, in invoke
    result = self.send(soapenv)
  File "/home/ubuntu/venv/local/lib/python2.7/site-packages/suds/client.py", line 613, in send
    reply = self.options.transport.send(request)
  File "/home/ubuntu/venv/local/lib/python2.7/site-packages/googleads/util.py", line 98, in PatchedHttpTransportSend
    fp = self.u2open(u2request)
  File "/home/ubuntu/venv/local/lib/python2.7/site-packages/suds/transport/http.py", line 132, in u2open
    return url.open(u2request, timeout=tm)
  File "/usr/lib/python2.7/urllib2.py", line 410, in open
    response = meth(req, response)
  File "/usr/lib/python2.7/urllib2.py", line 523, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib/python2.7/urllib2.py", line 448, in error
    return self._call_chain(*args)
  File "/usr/lib/python2.7/urllib2.py", line 382, in _call_chain
    result = func(*args)
  File "/usr/lib/python2.7/urllib2.py", line 531, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
urllib2.HTTPError: HTTP Error 500: Internal Server Error`

I have not changed the code at all.

**Comment by AlanVerbner** [comment]
@mfaraji which version are you using? The following works for me:

```
bash-4.3# pip show googleads                        
---
Metadata-Version: 1.1
Name: googleads
Version: 4.7.0
Summary: Google Ads Python Client Library
```

But `5.0.0` returns the same error you posted.

There was a release recently. Check out the changes stated [here](https://github.com/googleads/googleads-python-lib/releases).

hope that helps!

**Comment by mfaraji** [comment]
I think mine is the recent one.

---

Name: googleads

Version: 5.0.0

Location: /home/ubuntu/venv/lib/python2.7/site-packages

Requires: httplib2, oauth2client, suds-jurko, pysocks, pytz, PyYAML,
xmltodict

**Comment by AlanVerbner** [comment]
If you can and don't mind running an old version, try doing this:

`pip uninstall googleads`
`pip install 'googleads==4.7.0'`

That's how we fixed our problem.

**Comment by mfaraji** [comment]
Here is a new error:

Traceback (most recent call last):

  File "audience.py", line 85, in <module>

    get_first_party_audience_segments(dfp_client)

  File "audience.py", line 43, in get_first_party_audience_segments

    statement.ToStatement())

  File
"/home/ubuntu/venv/local/lib/python2.7/site-packages/googleads/common.py",
line 647, in MakeSoapRequest

    raise e

suds.WebFault: Server raised fault:
'[AuthenticationError.NETWORK_CODE_REQUIRED @ ]'



Thanks,

**Comment by mfaraji** [comment]
I am running your example code.

**Comment by thangduo** [comment]
The [AuthenticationError.NETWORK_CODE_REQUIRED ](https://developers.google.com/doubleclick-publishers/docs/troubleshooting#AuthenticationError.NETWORK_CODE_REQUIRED) indicates that no network code was provided.

The example code, by default, looks for the `googleads.yaml` file at your home directory, which must contain the network code to service your SOAP requests. You can download the [model of the `googleads.yaml` file](https://github.com/googleads/googleads-python-lib/blob/master/googleads.yaml), and then fill in the network code on [line 50](https://github.com/googleads/googleads-python-lib/blob/master/googleads.yaml#L50).

Network code: https://docs.google.com/a/google.com/drawings/d/1Ufh0s1S0h9TZ6bumEFU65P0_93_Rb2kqkTyhB6htpLc/edit?usp=sharing

**Comment by thangduo** [comment]
@mfaraji Does specifying network code resolve your issue? If not, do you see any other error messages?

**Comment by mfaraji** [comment]
Yes, actually it works. Is there any service to get the list of countries
for geo-tagging from DFP?

**Comment by thangduo** [comment]
Great to hear that it works for you.

Here's how you can build your geo targeting list:
https://developers.google.com/doubleclick-publishers/docs/reference/v201605/PublisherQueryLanguageService


Thanks

**Comment by thangduo** [comment]
Just FYI, here's a video about custom targeting: https://youtu.be/NrbyhCFCkW8

**Comment by vtsao** [comment]
We're putting in a fix so that the actual error is surfaced and not an HTTP 500 error for cases like this, so re-opening this issue to track that.

**Comment by thangduo** [comment]
Resolved in 5.1.0 release
